### PR TITLE
{AKS} Extend `--data-collection-settings` with containerlogv2

### PR DESCRIPTION
**Pull request by wanlonghenry**
**Related command**

**Description**
Extend DataCollectionSettings JSON for containerlogv2

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by wanlonghenry** [comment]
image

**Review by ganga1980 — APPROVED** [review]

**Comment by wanlonghenry** [comment]
@zhoxing-ms @yonzhan  could you review this PR when available, thank you!

File: src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py [hunk 1]
```diff
@@ -599,9 +599,9 @@ def validate_data_collection_settings(dataCollectionSettings):
         if isinstance(namspaces, list) is False:
             raise InvalidArgumentValueError('namespaces must be an array type')
     if 'enableContainerLogV2' in dataCollectionSettings.keys():
-        enableContainerLogV2Value = dataCollectionSettings["enableContainerLogV2"].lower()
-        if enableContainerLogV2Value not in ["true", "false"]:
-            raise InvalidArgumentValueError('enableContainerLogV2Value value MUST be either true or false')
+        enableContainerLogV2Value = dataCollectionSettings["enableContainerLogV2"]
+        if not isinstance(enableContainerLogV2Value, bool):
+            raise InvalidArgumentValueError('enableContainerLogV2Value value MUST be either True or False')
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py:604]: Maybe lowercase is better as user provides the config in json format.

Comment (wanlonghenry) [reply src/azure-cli/azure/cli/command_modules/acs/addonconfiguration.py]: changed

**Comment by wanlonghenry** [comment]
@zhoxing-ms  @yonzhan @FumingZhang  addressed the comment, could you check again, thank you!

**Comment by zhoxing-ms** [comment]
@wanlonghenry  This is a known issue related to pydantic https://github.com/pydantic/pydantic/issues/6361, we are tracking it

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**